flask/app.py

```diff
@@ -1370,6 +1390,33 @@ def url_defaults(self, f):
         self.url_default_functions.setdefault(None, []).append(f)
         return f
 
+    def _find_error_handler(self, e):
+        """Finds a registered error handler for the request’s blueprint.
+        If neither blueprint nor App has a suitable handler registered, returns None
+        """
+        exc_class, code = self._get_exc_class_and_code(type(e))
+        
+        def find_superclass(d):
+            if not d:
+                return None
+            for superclass in exc_class.mro():
+                if superclass is BaseException:
+                    return None
+                handler = d.get(superclass)
+                if handler is not None:
+                    d[exc_class] = handler  # cache for next time exc_class is raised
```

Comment:
>and the more efficient algorithm may be to statically include all exceptions in the mapping upon app start. so basically replacing my JIT cache with an AOT cache :smile:

That's impossible, we can't reach subclasses from parent classes.

Honestly I think instead of proving that additional code doesn't cause problems, it should be proven that the absence of the code doesn't cause problems. IOW, the right for the cache to exist doesn't lie in "it doesn't cause any problems", but rather "it improves performance". And that has to be proven instead of saying "why not".